src/gui/sdlmain.cpp

```diff
@@ -1355,7 +1355,19 @@ static SDL_Window* SetWindowMode(const RenderingBackend rendering_backend,
 	// latency (and not the rendering pipeline).
 	render_pacer->Reset();
 
+#if C_OPENGL
+	const auto rendering_backend_changed = (sdl.rendering_backend !=
+	                                        rendering_backend);
+#endif
+
 	sdl.rendering_backend = rendering_backend;
+
+#if C_OPENGL
+	if (rendering_backend_changed) {
+		RENDER_Reinit();
```

Comment:
So it trips over here, because deep inside `GFX_SetSize()` is called which expects some values to be already populated (like maximum texture size). But moving it to the end of `set_output()`, when everything is setup and should be mostly ready to go doesn't help, as now it trips inside shader building... because the context gets destroyed along the way.

This place is cursed, I'm telling you. Digging further.

Reply:
Scratch that one, I was just overzealously cleaning up resources to right the wrongs. Moving `RENDER_Reinit()` to the end of `set_output()` seems to be enough to make it functional for now.

Reply:
Thanks @GranMinigun , I messed around with it for like 15 minutes before going to bed but got frustrated with the sdlmain mess. Glad you found a workable solution and I'm really looking forward to your SDL rework hitting main!